app/service/update-appeal-service.ts

```diff
@@ -310,23 +373,32 @@ export default class UpdateAppealService {
       }
     }
 
-    caseData.timeExtensions = appeal.previousAskForMoreTime || [];
+    caseData.timeExtensions = [];
+    const previousAskForMoreTimes = appeal.previousAskForMoreTime;
 
+    if (previousAskForMoreTimes) {
+      previousAskForMoreTimes.forEach(askForMoreTime => {
+        this.addCcdTimeExtension(askForMoreTime, appeal, caseData);
+      });
+    }
     const askForMoreTime = appeal.askForMoreTime;
-    if (askForMoreTime && askForMoreTime.status === 'submitted') {
+    if (askForMoreTime && askForMoreTime.reason) {
       this.addCcdTimeExtension(askForMoreTime, appeal, caseData);
     }
 
     return caseData;
   }
 
   private addCcdTimeExtension(askForMoreTime, appeal, caseData) {
+    if (askForMoreTime.status === 'inProgress') {
+      askForMoreTime.status = 'submitted';
+    }
```

Comment:
InProgress is different to submitted. inProgress is when the frontend user is filling in the time extension submitted when they have clicked submit on the CYA page so don't think this is correct.